lib/profiler/get_tick.h

```diff
@@ -67,6 +67,19 @@ get_tick(void)
     return regs;
 }
 
+#elif defined(__APPLE__)
+#include <mach/mach_time.h>
+
+static __inline hrtime_t
+get_tick(void)
+{
+    static mach_timebase_info_data_t base;
+    uint64_t ticks;
+    (void) mach_timebase_info (&base);
+    ticks = mach_absolute_time ();
+    return (hrtime_t)((ticks * base.numer) / base.denom);
```

Comment:
Please avoid division by zero. It might happen if mach_timebase_info() fails, but I probably would not restrict the check to that specific case.

Reply:
mach_timebase_info in fact never fails even tough it does return a state.

Reply:
FYI: Apple [documentation](https://developer.apple.com/documentation/driverkit/3433733-mach_timebase_info) implies otherwise. Perhaps it never fails today but might start failing tomorrow?

Reply:
Ah, I see that you removed that function. Never mind then. I am resolving this thread.